compiler/src/dotty/tools/io/PlainFile.scala

```diff
@@ -78,7 +78,7 @@ class PlainFile(val givenPath: Path) extends AbstractFile {
   }
 
   /** Is this abstract file a directory? */
-  def isDirectory: Boolean = givenPath.isDirectory
+  val isDirectory: Boolean = givenPath.isDirectory
```

Comment:
This is not the best approach. The `givenPath` object is a `dotty.tools.io.Path`. In theory, directories should be `dotty.tools.io.Directory`. Therefore we should optimize `dotty.tools.io.Directory` instead.

```diff
class Directory(jpath: JPath) extends Path(jpath) {
+  override def isDirectory: Boolean = true
+  override def isRegularFile: Boolean = false
  ...
}
```

And we can do the same for `dotty.tools.io.File`

```diff
class Directory(jpath: JPath) extends Path(jpath) {
+  override def isRegularFile: Boolean = true
+  override def isDirectory: Boolean = false
  ...
}
```

Reply:
that doesn't work for some reason
i could cache in dotty.tools.io.Path instead

Reply:
hm actually caching might break something as both dotty.tools.io.AbstractFile and dotty.tools.io.Path are mutable

Reply:
updated pr